### PR TITLE
feat(datadog)!: Upgrade to SDK V3 Arrow native

### LINKED ISSUE #10738: feat: Migrate `plugin/source/datadog` to `github.com/cloudquery/plugin-sdk/v3`
### Which problem is this feature request solving?

Part of #10711

### Describe the solution you'd like

Upgrade the code.

### Pull request (optional)

- [X] I can submit a pull request

### PR DESCRIPTION
**Pull request by yevgenypats**
Closes #10738 

BEGIN_COMMIT_OVERRIDE
feat: Update to use [Apache Arrow](https://arrow.apache.org/) type system (#10976)

BREAKING-CHANGE: This release introduces an internal change to our type system to use [Apache Arrow](https://arrow.apache.org/). This should not have any visible breaking changes, however due to the size of the change we are introducing it under a major version bump to communicate that it might have some bugs that we weren't able to catch during our internal tests. If you encounter an issue during the upgrade, please submit a [bug report](https://github.com/cloudquery/cloudquery/issues/new/choose). You will also need to update destinations depending on which one you use:
- Azure Blob Storage >= v3.2.0
- BigQuery >= v3.0.0
- ClickHouse >= v3.1.1
- DuckDB >= v1.1.6
- Elasticsearch >= v2.0.0
- File >= v3.2.0
- Firehose >= v2.0.2
- GCS >= v3.2.0
- Gremlin >= v2.1.10
- Kafka >= v3.0.1
- Meilisearch >= v2.0.1
- Microsoft SQL Server >= v4.2.0
- MongoDB >= v2.0.1
- MySQL >= v2.0.2
- Neo4j >= v3.0.0
- PostgreSQL >= v4.2.0
- S3 >= v4.4.0
- Snowflake >= v2.1.1
- SQLite >= v2.2.0

END_COMMIT_OVERRIDE

### CODE REVIEW

**Comment by yevgenypats** [comment]
Closes https://github.com/cloudquery/cloudquery/issues/10738

File: plugins/source/datadog/go.mod [hunk 1]
```diff
@@ -4,8 +4,10 @@ go 1.19
 
 require (
 	github.com/DataDog/datadog-api-client-go/v2 v2.9.0
+	github.com/apache/arrow/go/v13 v13.0.0-20230509040948-de6c3cd2b604
 	github.com/cloudquery/plugin-pb-go v1.0.8
 	github.com/cloudquery/plugin-sdk/v2 v2.7.0
```
Comment (candiduslynx) [inline plugins/source/datadog/go.mod:9]: this shouldn't be there (only as indirect). You need to search where you left v2 reference

File: plugins/source/datadog/resources/services/monitors/monitor_downtimes.go [hunk 2]
```diff
@@ -17,104 +19,103 @@ func MonitorDowntimes() *schema.Table {
...
 			{
 				Name:     "start",
-				Type:     schema.TypeInt,
+				Type:     arrow.PrimitiveTypes.Int64,
 				Resolver: schema.PathResolver("Start"),
 			},
 			{
 				Name:     "timezone",
-				Type:     schema.TypeString,
+				Type:     arrow.BinaryTypes.String,
```
Comment (candiduslynx) [inline plugins/source/datadog/resources/services/monitors/monitor_downtimes.go:107]: gen header should be removed?

File: plugins/source/datadog/resources/services/roles/role_permissions.go [hunk 3]
```diff
@@ -4,9 +4,11 @@ import (
 	"context"
 
 	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/apache/arrow/go/v13/arrow"
 	"github.com/cloudquery/cloudquery/plugins/source/datadog/client"
-	"github.com/cloudquery/plugin-sdk/v2/schema"
-	"github.com/cloudquery/plugin-sdk/v2/transformers"
+	"github.com/cloudquery/plugin-sdk/v2/types"
```
Comment (candiduslynx) [inline plugins/source/datadog/resources/services/roles/role_permissions.go:9]:
```suggestion
	"github.com/cloudquery/plugin-sdk/v3/types"
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**